tree/ntupleutil/v7/src/RNTupleImporterCLI.cxx

```diff
@@ -0,0 +1,205 @@
+/// \file RNTupleImporterCLI.cxx
+/// \ingroup NTuple ROOT7
+/// \author Florine de Geus <florine.de.geus@cern.ch>
+/// \date 2023-10-25
+/// \warning This is part of the ROOT 7 prototype! It will change without notice. It might trigger earthquakes. Feedback
+/// is welcome!
+
+/*************************************************************************
+ * Copyright (C) 1995-2023, Rene Brun and Fons Rademakers.               *
+ * All rights reserved.                                                  *
+ *                                                                       *
+ * For the licensing terms see $ROOTSYS/LICENSE.                         *
+ * For the list of contributors see $ROOTSYS/README/CREDITS.             *
+ *************************************************************************/
+
+#include "ROOT/RNTupleImporterCLI.hxx"
+#include "ROOT/RNTupleImporter.hxx"
+
+#include "TROOT.h"
+
+#include <iostream>
+
+using namespace ROOT::Experimental::RNTupleImporterCLI;
+
+namespace {
+const auto usageText = "Usage:\n"
+                       " ttree2rntuple (--ttree|-t) name\n"
+                       "               (--infile|-i) path\n"
+                       "               (--outfile|-o) path\n"
+                       "               [(--rntuple|-r) name]\n"
+                       "               [(--compression|-c) compression]\n"
+                       "               [--unzipped-page-size size]\n"
+                       "               [--zipped-cluster-size size]\n"
+                       "               [--max-unzipped-cluster-size size]\n"
+                       "               [--convert-dots]\n"
+                       "               [(--verbose|-v)]\n"
+                       " ttree2rntuple [--help|-h]\n\n";
+
+const auto argUsageText =
+   "ttree2rntuple: a utility CLI for converting ROOT TTrees to ROOT RNTuples.\n"
+   "\n"
+   "Required arguments:\n"
+   "  --ttree, -t name\n"
+   "    The name of the source TTree to convert to RNTuple.\n"
+   "  --infile, -i path\n"
+   "    The path to the ROOT file that contains the source TTree.\n"
+   "  --outfile, -o path\n"
+   "    The path to the ROOT file to write the target RNTuple to. This file does not have to exist yet.\n"
+   "    This may be the same file as the input file. Note that in this case the name of the target RNTuple should\n"
```

Comment:
Since we advertise writing to the same file, do we test it?

Reply:
Yes, I added a test for this in the second patch set.